cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -354,17 +352,16 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		}
 
 		// In dry run only utilization is updated
-		calculateUnneededOnly := a.lastScaleUpTime.Add(a.ScaleDownDelayAfterAdd).After(currentTime) ||
+		calculateUnneededOnly := scaleDownForbidden ||
+			a.lastScaleUpTime.Add(a.ScaleDownDelayAfterAdd).After(currentTime) ||
 			a.lastScaleDownFailTime.Add(a.ScaleDownDelayAfterFailure).After(currentTime) ||
 			a.lastScaleDownDeleteTime.Add(a.ScaleDownDelayAfterDelete).After(currentTime) ||
-			schedulablePodsPresent ||
-			scaleDown.nodeDeleteStatus.IsDeleteInProgress() ||
-			allPendingPodsToHelpAreNew
+			scaleDown.nodeDeleteStatus.IsDeleteInProgress()
 
 		glog.V(4).Infof("Scale down status: unneededOnly=%v lastScaleUpTime=%s "+
-			"lastScaleDownDeleteTime=%v lastScaleDownFailTime=%s schedulablePodsPresent=%v isDeleteInProgress=%v",
```

Comment:
I admit that now the information is spread a bit - we log "Schedulable pods present" earlier (and on higher log level) if we fall into this case. Lack of this log indicates "all pods are new" but it might not be obvious to the users which don't look deeper into the code.
Thinking about it now I'd rather add logs when the decision happens and maybe even remove the log you've highlighted since it contains a lot of information which might not have been used at all.

Reply:
Oops, I thought for a moment this is status and not log line. I agree we probably don't need to worry about removing it then. I'd still consider adding it to status as unlike logs, this is visible to GKE users.